config/http.go

```diff
@@ -0,0 +1,50 @@
+package config
+
+import (
+	"crypto/tls"
+	"net/http"
+	"sync/atomic"
+
+	"github.com/pomerium/pomerium/internal/log"
+	"github.com/pomerium/pomerium/pkg/cryptutil"
+)
+
+type httpTransport struct {
+	underlying *http.Transport
+	transport  atomic.Value
+}
+
+// NewHTTPTransport creates a new http transport. If CA or CAFile is set, the transport will
+// add the CA to system cert pool.
+func NewHTTPTransport(src Source) http.RoundTripper {
+	t := new(httpTransport)
+	t.underlying, _ = http.DefaultTransport.(*http.Transport)
```

Comment:
Can this be `http.DefaultTransport.Clone()` ?

Reply:
Ahh, I see. 
https://github.com/pomerium/pomerium/pull/1576/files#diff-f6d92a8f39b323b40cde569ed5f78ba3e1704e2c5976fa7b9046f8b093fef485R32